### PR TITLE
Update the Steam provider to run the Authenticated event when the userinfo endpoint is null or when no application key was set

### LINKED ISSUE #51: OnAuthenticated event not firing - Can't set Identity
There is an issue when setting Identity.
image
A breakpoint in the OnAuthenticated method is never hit, so I really can't set the Identity.
Authentication happens just fine, and CallbackPath is showed right after, as supposed.
I noticed you @kelvinmac has the same issue. Don't know if your breakpoints are hit though.
Any suggestions?
Or just any suggestion for getting Steam ID?
I'm kinda stuck because I can't retrieve any user data.

Thanks.

### PR DESCRIPTION
**Pull request by kevinchalet**
Fixes https://github.com/aspnet-contrib/AspNet.Security.OpenId.Providers/issues/51.

@martincostello I think we should ship this fix as part of 2.1.0. Any objection?

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -4,6 +4,7 @@
   <Import Project="version.props" />
 
   <PropertyGroup>
+    <LangVersion>preview</LangVersion>
```
Comment (martincostello) [inline build/common.props]: Looks like the CI doesn't like this for some reason.

Comment (kevinchalet) [reply build/common.props]: Yeah, I guess that's because it requires a 3.0 SDK 😢

**Comment by martincostello** [comment]
I just checked the diff between 2.0.0 and dev, and everything's all infrastructure stuff, so this fix would be the only reason to do a 2.1.0 (or 2.0.1?) anyway.

**Comment by kevinchalet** [comment]
Yep. The other reason is that we've always kept the OAuth/OpenID versions in sync 😄

**Review by martincostello — APPROVED** [review]

**Comment by kevinchalet** [comment]
@martincostello thanks!

**Pull request merged**